### DISCUSSION TITLE
Including package docs

### DISCUSSION THREAD

**Discussion by mkschulze** [Humans-of-Julia/HoJBot.jl#25] [Ideas]
Hello,

as we have the docs functionality within our bot, it would be interesting to see which packages you would like to have available.

Here is my personal list:

- all HoJ packages
- HTTP.jl
- Genie.jl
- Stipple.jl
- DataStructures.jl
- CSV.jl
- DataFrames.jl
- Documenter.jl

**Comment by Ved-Mahajan** [comment]
Here are a few I feel are important:

Plots.jl
DifferentialEquations.jl
Flux.jl

**Comment by tk3369** [comment]
I'm unsure if adding all these dependencies is a good idea because we would have to import these packages even though the bot is only using the documentation but nothing else. It is also an unbounded problem because the number of packages will grow over time and it is not sustainable.

However, I don't have a good alternative to suggest either. Let's keep the discussion going and think about how to implement properly at the same time.

**Reply by mkschulze** [reply]
We could also benefit from including the packages in the bot when we integrate the Julia REPL maybe. Loading times shouldn't be a problem with 1.6 any more I guess?

**Reply by tk3369** [reply]
The issue is that it will consume memory for no reason other than fetching doc strings.

**Reply by mkschulze** [reply]
Is it hosted by us or from another Discord server eventually? I mean does it cost somebody's money or is the memory consumption on the client pc using our server?

**Reply by mkschulze** [reply]
If we can use the packages then in the REPL to show things in helping scenarios on the server it's worth it I think. Otherwise, I don't know, maybe we could web scrape the docs to achieve the same?

**Reply by rmsrosa** [reply]
The Discord bot works as a bridge (I don't know the technical term) between our Discord server and the server running the julia script, which currently is being hosted in @Azzaare 's server. It is not free and it has limited resources, so I don't think it is a good idea to keep adding packages just like that.

However, we could, in principle, create a database of docstrings from several different packages and store that. We would just need to update it regularly.

**Reply by Azzaare** [reply]
I don't mind giving a try to loading a huge list of packages and see how the server handles it.
Also, I don't mind upgrading the server if necessary.

**Reply by mkschulze** [reply]
Ok, then we might give it a try then I suppose?

**Reply by Azzaare** [reply]
We have a hojbot-dev user available, if we set it up, we can make the tests easily and the memory it eats.
